### PR TITLE
Add used memory amount to CircuitBreakingException message (#22521)

### LINKED ISSUE #22521: CircuitBreakingException message should contain the amount of used memory
Currently the message for CircuitBreakingException can be confusing. It only displays the limit of the circuit breaker but not the amount that is actually used. Example:

```
Caused by: org.elasticsearch.common.breaker.CircuitBreakingException: [parent] Data too large, data for [<transport_request>] would be larger than limit of [1491035750/1.3gb]
        at org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService.checkParentLimit(HierarchyCircuitBreakerService.java:211) ~[elasticsearch-5.0.2.jar:5.0.2]
```

This can lead to the impression that the request would need 1.3gb alone. It would be nice if the totalUsed could be included in the message to make it clear that the memory is potentially consumed by other requests. (totalUsed is available in checkParentLimit)

### PR DESCRIPTION
**Pull request by bumbu**
Closes #22521 

Tests don't check for the amount of used memory because that will tie-up the tests to the implementation. 
From what I saw there are no tests to check that the used memory is calculated correctly, but it may be just my lack of knowledge of the codebase. Also it is out of the scope of original ticket.

### CODE REVIEW

**Comment by dakrone** [comment]
@bumbu thanks for this! This looks pretty good, could you add a unit test to `MemoryCircuitBreakerTests` with exact numbers so we can assert the message looks correct? (I understand you can't easily add it to the integration tests since the desired memory is not constant)

**Comment by bumbu** [comment]
Hi @dakrone,

I added an assertion to `MemoryCircuitBreakerTests` to check for the numbers returned in the message. I didn't hard-code the number, instead I computed it right before the assertion. Didn't want to hard-code the number as how it is currently done gives more insight on what that number is. 

Also I tested only for the first part of the number value (aka testing only for `1024` from `[1024/1k]`) as I didn't want to add one more dependency for computing the byte size value. The assertion should cover the message pretty well as the same number as not formatted and formatted byte size value. But I can easily add that if it will bring more value. 

I squashed the commit.

File: core/src/test/java/org/elasticsearch/common/breaker/MemoryCircuitBreakerTests.java [hunk 1]
```diff
@@ -235,6 +235,9 @@ public void testConstantFactor() throws Exception {
             fail("should never reach this");
         } catch (CircuitBreakingException cbe) {
             assertThat("breaker was tripped exactly twice", breaker.getTrippedCount(), equalTo(2L));
+
+            long newUsed = (long)(breaker.getUsed() * breaker.getOverhead());
+            assertThat(cbe.getMessage().contains("would be [" + newUsed "/"), equalTo(true));
```
Comment (dakrone) [inline core/src/test/java/org/elasticsearch/common/breaker/MemoryCircuitBreakerTests.java]: This is missing a `+` between `newUsed` and `"/"`, so it's not compiling currently

**Review by dakrone — CHANGES_REQUESTED** [review]
It looks like there's a compilation error, @bumbu otherwise this looks good, can you fix the error and then I'll merge this?

File: core/src/test/java/org/elasticsearch/common/breaker/MemoryCircuitBreakerTests.java (hunk 1 above)
Comment (bumbu) [inline core/src/test/java/org/elasticsearch/common/breaker/MemoryCircuitBreakerTests.java]: My bad. I was sure that it's not possible to introduce bugs with such a small change... Lesson learned.

**Comment by bumbu** [comment]
Hi @dakrone.
Sorry for the delay, I was on holidays. 
I fixed the bug, run tests, squashed commits and rebased from master. 
Please check.

**Comment by dakrone** [comment]
@elasticmachine test this please

**Comment by dakrone** [comment]
Thanks @bumbu, I've merged this!

**Pull request merged**